src/server/games.service.js

```diff
@@ -0,0 +1,161 @@
+// Games service
+
+const redis = require('redis');
+const User = require('./user.class');
+const Game = require('./game.class');
+const Redis = require('./redis.service');
+const Server = require('./server.service');
+const Users = require('./users.service');
+
+// The class is given when something requires this module
+module.exports = class Games {
+
+    static instance = null;
+
+    constructor() {
+    }
+
+    static init() {
+        if (Games.instance == null) {
+            Games.instance = new Games();
+        }
+    }
+
+    static getInstance() {
+        if (Games.instance == null) {
+            Games.init();
+        }
+        return Games.instance;
+    }
+
+    static async get(gameID) {
+        try {
+            let gameJSON = await Redis.get(gameID);
+            if (gameJSON) {
+                return JSON.parse(gameJSON);
+            }
+        } catch (e) {
+            console.error(e);
+        }
+        return undefined;
+    }
+
+
+    /**
+     * Get the phrases for the game
+     */
+    static async phrases(language) {
+
+        //TODO get phrases from the database
+
+        let phrases = [
+            "Mock phrase _____ round 1",
+            "Mock phrase _____ round 2",
+            "Mock phrase _____ round 3",
+        ];
+
+        return phrases;
+    }
+
+    /**
+     * Create a new game and call the providede callback passing the created
+     * game object.
+     * */
+    static async create(room, cb, errCB) {
+        // Create new object
+        let redisIO = Redis.getIO();
+
+        if (!room || !room.ownerID || !room.users) {
+            if (errCB) {
+                errCB(new Error('Invalid room object'));
+            }
+            return undefined;
+        }
+
+        // Derive game ID from room ID
+        let gameID = 'game_' + room.id.substring(5);
+        let toWatch = room.users.concat(gameID, room.id);
+
+        async function transaction(attempts) {
+            // Watch to prevent conflicts
+            redisIO.watch(toWatch, async (watchErr) => {
+                try {
+                    if (watchErr) throw (watchErr);
+
+                    let exist = await Redis.exists(gameID);
+                    if (exist) {
+                        throw new Error(`Game ${gameID} already exists`);
+                    }
+
+                    let game = new Game(gameID);
+
+                    // TODO: get phrases ramdomly
+                    let allPromises = [];
+                    for (let player of room.users) {
+                        allPromises.push(Users.get(player));
+                    }
+
+                    //TODO provide language
+                    allPromises.push(Games.phrases(undefined));
+
+                    Promise.all(allPromises).then((values) => {
+
+                        // Extract the phrases from the end of the array
+                        game.phrases = values.pop;
+                        game.players = values;
```

Comment:
The `phrases` are being sent at the end of the player list, not sure if this is the problematic code:
```json
{
   "id":"room_35268",
   "users":[
      {
         "name":"uma",
         "id":"user_yyBPh7a1bwZWB_ARAAAP",
         "room":"room_35268"
      },
      {
         "name":"web1",
         "id":"user_QPMPwxVS_y_QaaMWAAAB",
         "room":"room_35268"
      },
      {
         "name":"user",
         "id":"user_LEA8j7v1OEXtY81VAAAB",
         "room":"room_35268"
      }
   ],
   "ownerID":"user_yyBPh7a1bwZWB_ARAAAP",
   "game":{
      "id":"game_35268",
      "players":[
         {
            "name":"uma",
            "id":"user_yyBPh7a1bwZWB_ARAAAP",
            "room":"room_35268"
         },
         {
            "name":"web1",
            "id":"user_QPMPwxVS_y_QaaMWAAAB",
            "room":"room_35268"
         },
         {
            "name":"user",
            "id":"user_LEA8j7v1OEXtY81VAAAB",
            "room":"room_35268"
         },
         [
            "Mock phrase _____ round 1",
            "Mock phrase _____ round 2",
            "Mock phrase _____ round 3"
         ]
      ],
      "personaVotes":[
         
      ],
      "answers":[
         
      ]
   }
}
```

Reply:
Hum, this is a bug, hahaha. I create an array of promises to use ``Promise.all()`` to wait for all to complete, which returns me an array of values. Probably something didn't work as I expected.